src/helper/mutations.ts

```diff
@@ -228,44 +159,108 @@ export function useCreateThreadMutation() {
 export function useCreateCommentMutation() {
   const auth = useAuth()
   const entityId = useEntityId()
-  const showToastNotice = useToastNotice()
+  const mutation = gql`
+    mutation createComment($input: ThreadCreateCommentInput!) {
+      thread {
+        createComment(input: $input) {
+          success
+        }
+      }
+    }
+  `
 
   const createCommentMutation = async function (
     input: ThreadCreateCommentInput
   ) {
-    const args = {
-      query: gql`
-        mutation createComment($input: ThreadCreateCommentInput!) {
-          thread {
-            createComment(input: $input) {
-              success
-            }
-          }
-        }
-      `,
-      variables: { input },
-    }
-
-    try {
-      const response = await authFetchThread(args, auth)
+    const success = await mutationFetch(auth, mutation, input)
 
-      if (response.thread.createComment?.success) {
-        return !!(await mutate(`comments::${entityId}`))
-      }
-    } catch (e) {
-      handleError(showToastNotice, e)
-      return false
-    }
+    if (success) await mutate(`comments::${entityId}`)
+    return success
   }
 
   return async (input: ThreadCreateCommentInput) =>
     await createCommentMutation(input)
 }
 
-function handleError(
-  showToastNotice: ReturnType<typeof useToastNotice>,
-  response?: object
-) {
-  console.log(response)
-  showToastNotice(`Das hat leider nicht geklappt…`, 'warning')
+type MutationInput =
+  | NotificationSetStateInput
+  | UuidSetStateInput
+  | ThreadCreateThreadInput
+  | ThreadSetThreadArchivedInput
+  | ThreadSetThreadStateInput
+  | ThreadSetCommentStateInput
+  | ThreadCreateCommentInput
+
+type ApiErrorType = 'UNAUTHENTICATED' | 'FORBIDDEN'
+type ErrorType = ApiErrorType | 'UNKNOWN'
+
+export interface ApiError extends GraphQLError {
+  extensions: {
+    code: ApiErrorType
+  }
+}
+
+export async function mutationFetch(
+  auth: React.RefObject<AuthPayload>,
+  query: string,
+  input: MutationInput,
+  isRetry?: boolean
+): Promise<boolean> {
+  if (auth.current === null) return handleError('UNAUTHENTICATED')
+
+  try {
+    return await executeQuery()
+  } catch (e) {
+    const error = (e as ClientError).response.errors?.[0] as
+      | ApiError
+      | undefined
+
+    const type = error ? error.extensions.code : 'UNKNOWN'
+
+    if (type === 'UNAUTHENTICATED' && !isRetry) {
+      await auth.current.refreshToken()
+      return await mutationFetch(auth, query, input, true)
+    }
+
+    return handleError(type)
+  }
+
+  async function executeQuery(): Promise<boolean> {
+    const client = new GraphQLClient(endpoint, {
+      headers: auth.current
+        ? {
+            'access-control-allow-origin': '*',
```

Comment:
Does this even do anything? AFAIK access-control-allow-origin is only relevant as a response header (i.e. for the server, not for the client)

Reply:
Should probably remove it in `createAuthAwareGraphqlFetch` as well then.